fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt

```diff
@@ -136,15 +136,45 @@ class ProductAnalysisMapperTest {
     }
 
     @Test
-    fun `WHEN grade, rating and highlights are all null THEN it is mapped to no analysis present`() {
+    fun `WHEN grade, rating and highlights are all null and it is initial analysis THEN it is mapped to no analysis present`() {
         val actual =
             ProductAnalysisTestData.productAnalysis(
                 grade = null,
                 adjustedRating = null,
                 highlights = null,
-            ).toProductReviewState()
+            ).toProductReviewState(isInitialAnalysis = true)
         val expected = ReviewQualityCheckState.OptedIn.ProductReviewState.NoAnalysisPresent()
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `WHEN grade, rating and highlights are all null and it is not initial analysis THEN not enough reviews card is visible`() {
+        val actual = ProductAnalysisTestData.productAnalysis(
+            grade = null,
+            adjustedRating = null,
+            highlights = null,
+        ).toProductReviewState(isInitialAnalysis = false)
+
+        val expected = ReviewQualityCheckState.OptedIn.ProductReviewState.Error.NotEnoughReviews
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `WHEN grade, rating and highlights are not all null and it is not initial analysis THEN it is mapped to AnalysisPresent`() {
+        val actual = ProductAnalysisTestData.productAnalysis(
```

Comment:
would there be any value in adding coverage for the other (6?) happy path combinations?

Reply:
yeah, definitely doesn't hurt, added now, makes refactoring/updating/maintaining in the future easier.